sdk/include/ndk/peb_teb.h

```diff
@@ -220,19 +244,49 @@ typedef struct STRUCT(_TEB)
     PTR(PVOID)             WOW32Reserved;
     LCID                   CurrentLocale;
     ULONG                  FpSoftwareStatusRegister;
+
+#if (NTDDI_VERSION >= NTDDI_WIN10) // since 10.0.10240.16384
+    PTR(PVOID)             ReservedForDebuggerInstrumentation[16];
+    PTR(PVOID)             SystemReserved1[38];
+#else
     PTR(PVOID)             SystemReserved1[54];
+#endif
     LONG                   ExceptionCode;
-#if (NTDDI_VERSION >= NTDDI_LONGHORN)
-    PTR(struct _ACTIVATION_CONTEXT_STACK*) ActivationContextStackPointer;
-    UCHAR                  SpareBytes1[0x30 - 3 * sizeof(PTR(PVOID))];
+#ifdef _STRUCT64
+    UCHAR                  Padding0[4];
+#endif
+
+#if (NTDDI_VERSION >= NTDDI_WS03SP1)
+    PTR(PACTIVATION_CONTEXT_STACK) ActivationContextStackPointer;
+#else
+    STRUCT(ACTIVATION_CONTEXT_STACK) ActivationContextStack;
+#endif
+
+#if (NTDDI_VERSION >= NTDDI_WIN10) // since 10.0.9926.0
+    PTR(ULONG_PTR)         InstrumentationCallbackSp;
+    PTR(ULONG_PTR)         InstrumentationCallbackPreviousPc;
+    PTR(ULONG_PTR)         InstrumentationCallbackPreviousSp;
+  #ifdef _STRUCT64
     ULONG                  TxFsContext;
-#elif (NTDDI_VERSION >= NTDDI_WS03)
-    PTR(struct _ACTIVATION_CONTEXT_STACK*) ActivationContextStackPointer;
-    UCHAR                  SpareBytes1[0x34 - 3 * sizeof(PTR(PVOID))];
+    UCHAR                  InstrumentationCallbackDisabled;
+    UCHAR                  Padding1[3];
+  #else
+    UCHAR                  InstrumentationCallbackDisabled;
+    UCHAR                  SpareBytes[23];
+    ULONG                  TxFsContext;
+  #endif
+#elif (NTDDI_VERSION >= NTDDI_WIN7)
+    UCHAR                  SpareBytes[_SELECT3264(36, 24)];
+    ULONG                  TxFsContext;
+#elif (NTDDI_VERSION >= NTDDI_VISTA)
+    UCHAR                  SpareBytes1[_SELECT3264(36, 24)];
+    ULONG                  TxFsContext;
+#elif (NTDDI_VERSION >= NTDDI_WS03SP1)
+    UCHAR                  SpareBytes1[_SELECT3264(40, 28)];
 #else
-    ACTIVATION_CONTEXT_STACK ActivationContextStack;
-    UCHAR                  SpareBytes1[24];
+    UCHAR                  SpareBytes1[_SELECT3264(24, 0)];
```

Comment:
We've discussed this before, but I don't remember. Wouldn't a #error for the 64 bit case be better here?